### PR TITLE
Feature/add search functionality for models

### PR DESCRIPTION
**Pull request by wkirdp**
- Define filters for Billing Projects, Managed Groups, Workspaces and Accounts lists
- Use filters in BillingProjectList, ManagedGroupList, WorkspacesList, WorkspaceListByType, AccountList, AccountActiveList, and AccountInctiveList views.
- Add filter forms on billingproject_list, managedgroup_list, workspace_list and account_list templates
- Add tests

### CODE REVIEW

**Review by amstilp — CHANGES_REQUESTED** [review]
This looks good! I made some comments about things to change, mostly with tests, as well as some info about using the AccountAdapter. I'm also working on a way to make the filter form display more nicely on the page, so I'll see where I get with that.

**Comment by amstilp** [comment]
@wkirdp I played around with the filter form to get something that is a little less intrusive. Please see the changes I made to `BillingProjectFilter` and the `billingproject_list.html` template, and make the same changes for the other filters and list templates.

File: setup.cfg [hunk 1]
```diff
@@ -35,6 +35,7 @@ install_requires =
     google-auth >= 2.6
     fontawesomefree >= 6.1
     django-autocomplete-light >= 3.9
+    django-filter > 2.0
```
Comment (jmcarson) [inline setup.cfg:38]: Don't think this version is correct

Comment (amstilp) [reply setup.cfg:38]: @wkirdp Can you look into this? Did you mean > 23.0? >=23.0?

Comment (wkirdp) [reply setup.cfg:38]: @amstilp I pinned it to 23.2 and you said maybe not to pin to such a high version. Maybe just >2.0.  Did I misunderstand?  (After version 2.4, it jumped to 21.1)

Comment (jmcarson) [reply setup.cfg:38]: Sorry, I saw version 23.2 in requirements/base.txt and >2 in setup.cfg and thought it was unintentional. Ignore if that version makes sense.

**Review by amstilp — APPROVED** [review]

**Comment by amstilp** [comment]
CI issues seem to be intermittent and solved by retrying tox runs using the retry github action. The failures happen more often when there is no cache. It does not seem to be anything in the code.

**Pull request merged**